### ISSUE TITLE
Crossout (386180)

### ISSUE THREAD

**Issue by prototype99** [ValveSoftware/Proton#1937]
# Compatibility Report
- Name of the game with compatibility issues: crossout
- Steam AppID of the game: 386180

## System Information
- GPU: GTX 1070
- Driver/LLVM version: nvidia 396.54
- Kernel version: 4.16.0
- Link to full system information report as Gist: [Gist](https://gist.github.com/prototype99/fb562a72813537b6571076b3ac7c0b2d)
- Proton version: 3.16-4 beta

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-386180.log](https://github.com/ValveSoftware/Proton/files/2562764/steam-386180.log)

## Symptoms 
crashes after download. gjagent.exe stays open however, user must manually find and end process to relaunch. crash reporter does admittedly work though.

## Reproduction

simply install and try to run the game.

**Comment by MagmaiKH** [comment]
I can confirm I have the same issue.

**Comment by zzhiyi** [comment]
EasyAntiCheat

**Comment by HonkingGoose** [comment]
Implementation of some kind of whitelist/integration of EAC / anti-cheat systems into Proton is tracked here: #1468

**Comment by user1-github** [comment]
Just FYI, the game seems to work now. When I downloaded the game, it also downloaded the EAC Proton runtime and after I launched the game, I've seen a "success" message in EAC log.

Also, there are 2 EAC .so files in the game directory. Does that mean the developer himself enabled Proton support for the game? (So far I haven't seen any official word from the developer regarding Proton support).

**Comment by vars1ty** [comment]
> Just FYI, the game seems to work now. When I downloaded the game, it also downloaded the EAC Proton runtime and after I launched the game, I've seen a "success" message in EAC log.
> 
> Also, there are 2 EAC .so files in the game directory. Does that mean the developer himself enabled Proton support for the game? (So far I haven't seen any official word from the developer regarding Proton support).

The developers enabled support for it through EAC, their engine also has support for Linux.
Though, their implementation is **far** from being playable for some. I can enter games, last for a few seconds to minutes before it just crashes, have yet to find a solution for it.

Then there's micro-stuttering, which can be be somewhat resolved via VMTouch (Preload game-files into RAM)

**Comment by kisak-valve** [comment]
Crossout crashing with `stack smashing detected`

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6278.
@vars1ty posted on 2022-10-29T09:36:22:

# Compatibility Report
- Name of the game with compatibility issues: Crossout
- Steam AppID of the game: 386180

## System Information
- GPU: MSI GeForce RTX 3070 TI
- Driver/LLVM version: 520.56.06
- Kernel version: 6.0.5
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/vars1ty/d17023663ad7f346cb26ccfd66f842e3
- Proton version: 7.0-4

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
    - There is one, but it doesn't relate to this specific issue.
- [x] that I have checked whether there are updates for my system available.

Log: https://gist.github.com/vars1ty/3520dfe0b103578daa67522bfd0bcd63

## Symptoms
The game fails to launch 9,5/10 of the times, ending up with errors like `stack smashing detected`.
I have also tried running it with Wine-GE and various different Proton versions (+ GE), none which were stable.
The only thing that has worked at times was running it through Bottles, setting the synchronization to FSync. But even with that, it only lasts a handful of minutes before crashing again.

## Reproduction
1. Install Crossout
2. Launch the game
3. Watch it crash most of the time

### Edit
I've also gotten it confirmed by someone else that it occurs for them too, and not just me. Their distro is Kubuntu.

**Comment by JoshuaKimsey** [comment]
@kisak-valve (Apologies, just saw this was a transfer of @vars1ty's post ) I can confirm I get the `stack smashing detected - Terminated` error message as well.
(Where did you get your full set of logs from, out of curiosity?)

OS: Pop!_OS 22.04
CPU: i9-9900k
GPU: Nvidia RTX 2080
RAM: 64GB

I have a full terminal print out of it below from where I launched Steam via the terminal.

```bash
*** stack smashing detected ***: terminated
ThreadGetProcessExitCode: no such process 909409
ThreadGetProcessExitCode: no such process 909355
ThreadGetProcessExitCode: no such process 908887
ThreadGetProcessExitCode: no such process 908866
ThreadGetProcessExitCode: no such process 908858
ThreadGetProcessExitCode: no such process 908828
ThreadGetProcessExitCode: no such process 908825
ThreadGetProcessExitCode: no such process 908819
ThreadGetProcessExitCode: no such process 908815
pid 908821 != 908820, skipping destruction (fork without exec?)
Game process removed: AppID 386180 "/home/jkimsey/.steam/debian-installation/ubuntu12_32/reaper SteamLaunch AppId=386180 -- /home/jkimsey/.steam/debian-installation/ubuntu12_32/steam-launch-wrapper -- '/home/jkimsey/.secondary_drive/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/jkimsey/.secondary_drive/SteamLibrary/steamapps/common/Proton 7.0'/proton waitforexitandrun  '/home/jkimsey/.secondary_drive/SteamLibrary/steamapps/common/Crossout/launcher.exe' -steam", ProcID 909219 
ThreadGetProcessExitCode: no such process 909219
ThreadGetProcessExitCode: no such process 908837
ThreadGetProcessExitCode: no such process 908693
ThreadGetProcessExitCode: no such process 908692
Game 386180 created interface STEAMUSERSTATS_INTERFACE_VERSION012 / 
Game 386180 created interface SteamController008 / 
Game 386180 created interface SteamFriends017 / 
Game 386180 created interface SteamInput006 / 
Game 386180 created interface SteamUser021 / 
Game 386180 created interface SteamUser021 / User
Game 386180 created interface SteamUtils010 / 
Game 386180 method call count for IClientControllerSerialized::GetHandleForGamepadIndex : 29086
Game 386180 method call count for IClientUserStats::SetAchievement : 43
Game 386180 method call count for IClientUserStats::StoreStats : 1
Game 386180 method call count for IClientUserStats::RequestCurrentStats : 1
Game 386180 method call count for IClientUtils::RecordSteamInterfaceCreation : 10
Game 386180 method call count for IClientUtils::GetAppID : 15
Game 386180 method call count for IClientFriends::GetFriendByIndex : 33
Game 386180 method call count for IClientFriends::GetFriendCount : 1
Game 386180 method call count for IClientFriends::GetPersonaName : 2
Game 386180 method call count for IClientUser::GetAuthSessionTicket : 1
Game 386180 method call count for IClientUser::GetSteamID : 49
Game 386180 method call count for IClientUser::BLoggedOn : 29171
Uploaded AppInterfaceStats to Steam

(process:909286): GLib-GObject-CRITICAL **: 22:56:22.800: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
```

**Comment by vars1ty** [comment]
> (Where did you get your full set of logs from, out of curiosity?)

Via `PROTON_LOG=1`

**Comment by JoshuaKimsey** [comment]
@vars1ty Thanks. I have found one common connector here. On Proton 7.0-4 Stable, this event (shown below) always happens before a Stack Smashing crash happens:

```bash
644046.945:0350:0354:fixme:advapi:GetCurrentHwProfileA (0000000000112C58) semi-stub
644046.945:0350:0354:fixme:process:GetSystemRegistryQuota (0000000000112CD8, 0000000000112CDC) faking reported quota values
644047.206:0350:0560:trace:loaddll:free_modref Unloaded module L"C:\\windows\\system32\\rsaenh.dll" : builtin
644048.816:0350:0354:fixme:win:FlashWindowEx 000000000011C8B0 - semi-stub
*** stack smashing detected ***: terminated
```

It seems like that each time the `rsaenh.dll` is unloaded, a stack smashing event happens that crashes the game. This may be key for any devs who wanna try and figure out how to mitigate this issue. I've tried several of the Proton launch flags, with no avail. The crash log above was captured with `PROTON_NO_ESYNC=1`, so that's why it's a bit more messy, but has more info (to me it looks like there is more info).

But yeah, hopefully with this info, a fix/workaround can be found. I'm actually not sure why Proton Experimental isn't working, I need to do more testing on that.

**Comment by JoshuaKimsey** [comment]
Update: just tested with Proton Experimental with the same settings as above, and it's definitely an EAC issue with regards to that. Here's the crash log from just before the crash at startup:

```bash
645163.825:0518:0540:trace:seh:RtlRestoreContext returning to 000000007B6325FA stack 0000000011D5D640
645163.825:0518:0540:fixme:iphlpapi:NotifyAddrChange (Handle 0000000011D5E118, overlapped 000034B000304B10): stub
645163.826:0518:0540:trace:loaddll:build_module Loaded L"C:\\windows\\system32\\wlanapi.dll" at 0000000399110000: builtin
645163.826:0518:0540:fixme:wlanapi:WlanEnumInterfaces (0000000000000001, 0000000000000000, 0000000011D5D0B8) semi-stub
645163.831:0518:0540:fixme:winsock:setsockopt Ignoring SO_RANDOMIZE_PORT
645165.851:0390:0394:trace:loaddll:build_module Loaded L"Z:\\home\\southernwolf\\.secondary_drive\\SteamLibrary\\steamapps\\common\\Crossout\\EasyAntiCheat\\easyanticheat_x64.dll" at 00000002F6650000: builtin
*** stack smashing detected ***: terminated
pid 1367949 != 1367947, skipping destruction (fork without exec?)
```

So yeah, for whatever reason EAC seems to work ok on Proton 7.0-4 (assuming it's not causing the stack smashing there), but it crashes instantly on Proton Experimental.

**Comment by vars1ty** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/1937#issuecomment-1298272954

@JoshuaKimsey - Tested with 7.0-4 and crashes on it too, just that it takes a few minutes. Here's some info about [rsaenh](https://www.processlibrary.com/en/directory/files/rsaenh/20850/) though, might try and hand-replace it later today/tomorrow to see if it fixes anything.

**Comment by vars1ty** [comment]
Alright I just tried to replace the DLL with one from a brand-new Windows 10 VM, it launched and seemed to work until it crashed. Gonna look for other solutions, though this seems a lot like it's something that either Easy AntiCheat messed up, or Targem really don't know how to code.

Here's a [bug report](https://community.gaijin.net/issues/p/crossout/i/FnAKPNFqxM1H) I've made on their tracker (don't expect them to care though), feel free to upvote it.

**Comment by JoshuaKimsey** [comment]
Cool, I can do that. It's unfortunate that didn't work, but I'm not wholly convinced it's EAC alone doing it. The reason being that last night I was able to play for over an hour straight, with no issues. But, later on when I relaunched the game, it crashed within 10 mins. I feel like if it was EAC, it would crash like it does with Proton Experimental, instantly. But, then maybe not...

IDK, hopefully either Gaijin can fix this, or perhaps the Proton devs can find a work around for it.

**Comment by vars1ty** [comment]
Yeah experienced the randomness of it working for hours and then randomly just deciding not to work.
I guess either wait for someone to *hopefully* find a fix for this, since Targem aren't really the most caring developers when it comes to Linux.

**Comment by vars1ty** [comment]
XO is having an update today apparently, so let's hope that they do something to fix it.
Will edit this message later when I know if it's been fixed or not.

**UPDATE**: Not fixed.

**Comment by JoshuaKimsey** [comment]
Dang it... Does it crash instantly or after a few mins?

**Comment by vars1ty** [comment]
@JoshuaKimsey - After a few minutes, like before.

Although I ***might*** have found a very hacky and weird way of mitigating some of the crashes, no promises though as I'll do some more testing (and I'll probably jinx myself for saying this).
Been in-game for I don't know, around 30 minutes? No crashes so far.

**Comment by vars1ty** [comment]
Alright restarted the game and it decided to start dying again, just like before.

**Comment by JoshuaKimsey** [comment]
Yeah. I had the same experience. Occasionally it won't crash and keeps going. Then it does nothing but crash...

**Comment by JoshuaKimsey** [comment]
@vars1ty What was the hacky way you found to make it work? Figured I could give it a try, if nothing else. I know some people on the Steam discussion forums are talking about Bottles and ProtonGE, but also seemed to have limited success, at best.

**Comment by Zykon88** [comment]
@JoshuaKimsey Normal works on Bottles + ProtonGE-38, but there is a problem after the update and this problem memory leak

**Comment by vars1ty** [comment]
@JoshuaKimsey - Bottles and ProtonGE only works for about 5 minutes, then it crashes.
The "hack" I did was by forcing DXVK to:
- Run DX11.0 (not 11.1 which XO defaults to)
- Spoof GPU PCI Device ID to something random
- Limit the FPS to 144
- Swap GPU Description to "AMD", which is also displayed in-game
- Enable some `nvapiHack` which reports NVIDIA GPUs as AMD ones
- Remove some DX11 barriers

Here's the config file: https://workupload.com/file/RMrBQ8CdZme
Environment variable: `DXVK_CONFIG_FILE=/home/USERNAME/dxvk.conf`
Runner: Soda 7.0-4
Ran through Bottles from Flatpak

**Comment by vars1ty** [comment]
Let me guess, no improvements on this?

**Comment by vars1ty** [comment]
For those still with this issue (perhaps you @Zykon88 and @JoshuaKimsey ?)
Tested playing for around 2h/3h today and no crashes so far (probs gonna jinx myself when I get on next time).
Seems like some update fixed it, here's the config I'm using:

- Bottles from the Arch AUR, for other distros: the non-flatpak version
- Synchronization on ESync for testing (switch to FSync if you can since it has better performance)
- Runner on **GE-Proton7-39**
- Preload Game Files turned on (package for doing this is `vmtouch`)
- Bottles & Steam Runtime turned off
- VKD3D on `vkd3d-proton-2.7`
- Latest Crossout Client + Launcher update
- No custom environment variables

Edit: Tested restarting the game between the 1h-mark, lasted for yet another hour afterwards until I closed the game.

**Comment by user1-github** [comment]
>     * VKD3D on `vkd3d-proton-2.7`

A bit off-topic regarding the latest issue, but I know that one of the recent updates introduced dx12, but I couldn't find any way to enable it. Since you mentioned vkd3d, did you find a way to enable dx12 in Crossout?

**Comment by vars1ty** [comment]
@user1-github - No, I just have all my runners set to the latest version in case there's any fixes

**Comment by JoshuaKimsey** [comment]
@vars1ty Ok, that's good to know. I haven't actually tried now in a week or so. What I'm gonna do is test with Proton Stable and Experimental first, then see if I need to explore Bottles and GE. According to your info, you have Bottles installed, but turned off?

**Comment by JoshuaKimsey** [comment]
@vars1ty I can confirm it is still broken on Proton 7.0-4 and Experimental. I'm now gonna go through and see what I can do with the info you gace.

**Comment by vars1ty** [comment]
@JoshuaKimsey No you have to use Bottles with latest Proton-GE from the Runners-tab (in this case, 39). The other runners only caused issues for me, so I guess it varies.

Steam Crossout though seems completely dead, as it refuses to even launch for me

**Comment by JoshuaKimsey** [comment]
@vars1ty Any chance you could create a tutorial for how to get this setup? I tried using Bottles, but I couldn't figure out how to get it working with Proton-GE inside of it. It didn't work running Steam through Bottles.

**Comment by vars1ty** [comment]
@JoshuaKimsey Yeah could do either later tonight, or in the morning

**Comment by vars1ty** [comment]
Here @JoshuaKimsey - https://streamable.com/i7ly53

**Comment by JoshuaKimsey** [comment]
@vars1ty Awesome! I'm trying this now to see if it'll work correctly. Good tutorial video.

**Comment by JoshuaKimsey** [comment]
@vars1ty Nope, it's still crashing for me. It won't launch without f-sync, and then it play's for maybe 5 mints top's before crashing. I can't usually make it through a single match. This sucks...

**Comment by vars1ty** [comment]
@JoshuaKimsey Which distro? Starting to sound like it's some outdated package/packages, since I just left Crossout untouched for days and updated my system before it started randomly working

**Comment by JoshuaKimsey** [comment]
@vars1ty Pop!_OS 22.04, with Nvidia hardware. It could be an out of date issue? But, I'm not sure. I have the latest Nvidia drivers that I'm aware of.

**Comment by vars1ty** [comment]
@JoshuaKimsey It's probably some outdated misc packages, I'm on Arch and ever since I randomly updated my PC a few days ago, XO has been working fine. Also had a friend try it since he was also crashing, same configuration on Bottles as me and 0 crashes.

**Comment by JoshuaKimsey** [comment]
@vars1ty That could be the case then. I may just need to wait and see if upstream updates fix the issues I'm having. Thanks for the help on this.

**Comment by vars1ty** [comment]
np, should probably be within a week or so (hopefully). Make sure you check for updates daily and keep on trying 👍🏽

**Comment by LordGrimmauld** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/1937#issuecomment-1312279214

This setup does work without issues for me so far, manjaro on unstable branch (so pretty much the same as the reported working arch default).
I also got latest Proton Stable working on steam by copying two missing DLLs from my old windows install to system32 of the compat wine prefix, but that was very crash happy for reasons i could not debug.
I'll document any changes to my current setup in here.
Also, the crash reporter seems to not like ProtonGE, i'll likely write a post launch script to just kill the crash reporter executable.

**Comment by vars1ty** [comment]
For those still with issues: Apparently the latest Proton version (`Proton 7.0.5`) fixes the crashes, although I don't run Crossout through Steam anymore so can't confirm but heard from a friend. Might be worth a shot trying via both Flatpak Steam and non-Flatpak.

**Comment by joni999** [comment]
Proton 7.0.5 worked for me - I have had no more crashes with it on my native Steam installation.
GE-Proton-7-41 on the other hand does have crashes.

**Comment by user1-github** [comment]
Haven't played Crossout in a while, but now when Proton 7.0-5 is out, I can confirm that it doesn't crash with this version. However, just like before, sometimes an error message pops up shortly after I start the game and it says that Crossout launcher has crashed. Now with Proton 7.0-5, every time the launcher starts, there's also an error message that Gaijin net agent (or whatever that tray thing is called) has crashed. This is of course not a very big deal because the game itself now works very well, but has anyone experienced these crashes?

**Comment by vars1ty** [comment]
@user1-github - If you are talking about the launcher crashing when exiting the game, yeah I get that too but it doesn't affect the game.
If it crashes prior to that, then no but it shouldn't be. Gaijin Updater crashing is normal but doesn't affect anything really.

**Comment by user1-github** [comment]
After about 3 weeks break, today I returned to the game and was greeted with another surprise: Now when I try to log into the game, I get a "Failed to initialize anti-cheat" error message.

This game has so many issues, I'm starting to lose my patience...

**Comment by user1-github** [comment]
Just a small tip if someone gets the "Failed to load anti-cheat" error in the game. It seems the game doesn't automatically download the Proton Easy Anticheat Runtime anymore, so if you get this error, just search for the runtime in your Steam library and manually install it.

**Comment by mathew2214** [comment]
it is impossible for me to sign into this game. after inputing my login and password, same i use for Gaijin website, i just get an error. same goes for signing in via steam. i cannot get past the login screen. 
i have reached out to Gaijin support to see if there is any issue with my account, but i have yet to receive any response. 
due to the fact i can sign in via their website, i assume there is no issue with my account, and that my inability to log into Crossout may be an issue with Proton.

**Comment by user1-github** [comment]
@mathew2214 What error do you get? If it's "Failed to load anti-cheat", see my comment above.

**Comment by mathew2214** [comment]
i can now sign in. Gaijin support discovered and corrected an issue with my account. but now i am getting anti-cheat errors. i have manually installed the Proton EAC runtime via the steam client. 

UPDATE: game is now playable by switching to Proton-Experimental

**Comment by vars1ty** [comment]
For those still experiencing issues, here's what seems to be the cause from a crash I had which for once gave some info on it:
```
18:50:14.825  WARNING| Unsupported driver version. Reason 'SwapChainCreation crash'
18:50:14.825  WARNING| Unsupported driver version. Reason 'Crash on cube texture bind on geometry stage'
```
tl;dr: Looks like it's a NVIDIA issue, to nobody's surprise.

**Comment by JoshuaKimsey** [comment]
@vars1ty Still getting what appears to be EAC errors on my end. That and graphics performance seems really bad.

**Comment by vars1ty** [comment]
@JoshuaKimsey - Try this config:
Bottles from Flatpak:
- Runner: Proton-GE 49
- GameScope: Enabled (Fullscreen) - `flatpak install gamescope`
- DXVK: dxvk-1.10.3 - v2.0 and higher crashes constantly for me, even with a new AMD GPU
- Sync: Fsync
- Feral GameMode & Preload Game Files: Enabled

Environment variables:
- PROTON_HEAP_DELAY_FREE: 1
- PROTON_NO_D3D10: 1
- WINE_DISABLE_VULKAN_OPWR: 1

**Comment by FlareFlo** [comment]
> @vars1ty Still getting what appears to be EAC errors on my end. That and graphics performance seems really bad.

Game barely manages to boot, crashes with stack-smashing before loading into lobby.

**Comment by Jimbolino** [comment]
cannot even get past the login screen

"Failed to initialize anti-cheat."

**Comment by vars1ty** [comment]
@Jimbolino - https://github.com/vars1ty/Linux-Fixes#proton-doesnt-work-with-eac

Overall this game is a huge hit-or-miss, a friend has it running flawlessly with an AMD GPU (which I switched to recently), same distro, same GPU drivers, etc. Only difference being the CPU and that's it unless there's some obscure package he's got that's missing for the rest of us.

**Comment by sergezl** [comment]
removing the lib32-amdvlk package helped me (Arch, Radeon RX 570)

**Comment by vars1ty** [comment]
For those still experiencing crashes and other issues, try this project I just made today.
https://github.com/vars1ty/CrossoutLD12
Currently it solves:
- Stack Smashing due to it not using DXVK directly, but rather VKD3D
- Limited to DirectX 11.x on Bottles and Steam
- (Potentially, not confirmed) Better performance due to less stuff going on in the background, removal of Gaijin and Targem's awful system tray and crash reporter

**Comment by coreybruce** [comment]
Game requires the following launch option to work on proton 8.0-3 or Experimental

```
PROTON_USE_WINED3D=1 %command%
```
Proton Log: https://gist.github.com/coreybruce/67ec24165d311ac0cb19db5463155511
Hardware info:  https://gist.github.com/coreybruce/3a7dd24ead8b58d8b85b4f94f5906a58

Also when you are shooting other vehicles you will get a bunch of graphical glitches al lover your screen as well as smoke 

![Screenshot from 2023-08-15 19-08-17](https://github.com/ValveSoftware/Proton/assets/8393780/0b41bc98-98a0-4613-aff4-02e4a025c0a3)

![Screenshot from 2023-08-15 18-56-25](https://github.com/ValveSoftware/Proton/assets/8393780/b5835558-a7c4-4091-a95e-cc27f378156e)

![Screenshot from 2023-08-15 19-09-29](https://github.com/ValveSoftware/Proton/assets/8393780/3a8f2706-0b62-458b-90a7-ce5155d57053)

**Comment by vars1ty** [comment]
> Game requires the following launch option to work on proton 8.0-3 or Experimental

@coreybruce - Yeah if you use DXVK, it requires it. The artifacts are from the fact that it isn't meant to be run that way.

Read my comment from before and use it, been using it myself for months and never had a single crash (or any artifacts) ever since.

**Comment by SiriusBlack-8** [comment]
@vars1ty I followed your guide included with the program from Github, but when I try to run the program with the launcher open, nothing ever happens. I triple checked that it is marked as executable and has the needed permissions.
Any help would be greatly appreciated.
Otherwise I use the launch Option "WINE_CPU_TOPOLOGY=4:0,1,2,3 DXVK_FRAME_RATE=60 %command%" without the quotes. This gets the game to actually launch without graphics issues, but it still crashes 3/4s of the way through a match.

**Comment by LordGrimmauld** [comment]
i did some experiments of my own, you can fix a lot of the crashes by setting dxvk shader compile to single-threaded (setting `dxvk.numCompilerThreads = 1` in your `dxvk.conf` file). But performance will suffer from that.

**Comment by SiriusBlack-8** [comment]
> i did some experiments of my own, you can fix a lot of the crashes by setting dxvk shader compile to single-threaded (setting `dxvk.numCompilerThreads = 1` in your `dxvk.conf` file). But performance will suffer from that.

@LordGrimmauld 
How badly does performance suffer? To the point you may as well not play anyway? Also have you tried other thread counts?

**Comment by LordGrimmauld** [comment]
its certainly playable, but you get some stutters. Notably when entering battle or going to garage or when exiting battle (which is probably when shaders do weird things), as well as a bit of startup delay

**Comment by Chris6ix** [comment]
I can confirm with the SteamDeck OLED. The game is listed as "playable" on SteamDeck, but it literally doesn't start. Depending on which version of proton you force it to use, it fails in different ways, but never works.
Proton versions I have tried and their results:
`Default`: Instant Black screen, nothing happening even after 2 minutes of waiting.
`Proton Experimental`: Shows the game's loading screen for 1 second, then closes the game and goes back to the library.
`Proton 9.0-2`: Crashes almost instantly, with the game's crash reporter coming up.
`Proton 8.0-5`: Crashes almost instantly, with the game's crash reporter coming up.
`Proton 7.0-6`: Gets into the login screen, but when trying to log in just gives the error "Failed to initialize the Anti Cheat, please restart or re-install the game"
`Proton 6.3-8`: Gets into the login screen, but when trying to log in just gives the error "Failed to initialize the Anti Cheat, please restart or re-install the game"

**Comment by mgnsk** [comment]
The game works with `GE-Proton9-9` on Steam Deck OLED.

**Comment by Chris6ix** [comment]
> The game works with `GE-Proton9-9` on Steam Deck OLED.

And how do I get `GE-Proton9-9`? It's not listed for me.

**Comment by mgnsk** [comment]
> > The game works with `GE-Proton9-9` on Steam Deck OLED.
> 
> And how do I get `GE-Proton9-9`? It's not listed for me.

In desktop mode, install ProtonUp-Qt from the Discover store and in there, install GE-Proton9-9. Currently the latest version is 9-10, I haven't tried that yet but also probably works.

**Comment by Chris6ix** [comment]
> In desktop mode, install ProtonUp-Qt from the Discover store and in there, install GE-Proton9-9. Currently the latest version is 9-10, I haven't tried that yet but also probably works.

Thank you very much!
Weirdly, though, only 9-9 works, and only in desktop mode. 9-10 in desktop mode and either 9-9 or 9-10 in gaming mode just make the game instantly crash with the game's crash reporter coming up.

There also seems to be an issue that I'm not sure if it's the game or protons fault, with broken texts like `QUESTS_HEADER_TUTORIAL_NAME_1` being displayed in a popup trying to tell me to do the tutorial.

After about an hour of playing, i can note that the game is also quite unstable. Lags, Freezes and crashes often mid game. Enabling Legacy GPU compatibility seems to have improved it a bit, but it's still an issue.

**Comment by mgnsk** [comment]
I've found a fix for the mid-game crashes on Steam Deck. Basically, the game needs to use WineD3D with Vulkan renderer instead of DXVK.


1) Go to Desktop Mode, open Steam and in game properties, change:
   * Under Compatibility, you may keep Force... unchecked or if you wish to pin the Proton version, check it and select the latest stable Proton version. Mine shows `Proton 9.0-2`. `Proton Experimental` also works.
   * Under General, set launch options: `PROTON_USE_WINED3D=1 WINE_CPU_TOPOLOGY=4:0,1,2,3 %command%`.

2) If this is a clean install, then In Steam, click PLAY to run the launcher once to let Steam create the wineprefix. You should also update the launcher if the Update Launcher button appears. When the launcher is updated, close it. You may also need to click Stop game in Steam.

3) Run these commands in the terminal to install protontricks and configure WineD3D to use the Vulkan renderer:

```
flatpak install --system com.github.Matoking.protontricks
```

```
flatpak run com.github.Matoking.protontricks 386180 renderer=vulkan
```

4) While in Desktop Mode, start the game. In the launcher settings, check **Legacy GPU compatibility** - otherwise the performance is abysmal with the WineD3D backend. Also due to WineD3D, the game only works in Desktop Mode. Enjoy!

**Comment by Chris6ix** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/1937#issuecomment-2282891273

Can confirm using the SteamDeck OLED with default Proton that this indeed fixes the crashes, but it makes the game stutter a bit more. But I suppose a bit worse performance is better than constant crashes, as it's at least playable now 😅

Thank you for finding this fix and sharing it :)

**Comment by OlivierFRDierick** [comment]
Moved from [steam-runtime](https://github.com/ValveSoftware/steam-runtime/issues/420)

Hello,

After a long hiatus, I can't seem to get the game CrossOut to run anymore.
I tried on Proton Experimental, 9.0-4, 8.0-5 and 7.0-6.
The launcher starts fine and can update the game. After pushing Play, there is the EAC progress bar and then a black loading screen that closes after a few seconds back to the steam library.
Proton log shows "stack smashing detected".

My GPU is nvidia RTX 2070 with proprietary driver 550.144.03.
I don't have old libGLX_nvidia.* files.
I don't use a container.

Regards.

[System information](https://gist.github.com/OlivierFRDierick/c455d96283cf6c733bda68d4f415db88)
[Proton Exp. log](https://gist.github.com/OlivierFRDierick/fffa098a43b302e68e1278d51259fbb9)

**Comment by woodbrettm** [comment]
Just did a fresh install Ubuntu 25.04, with Steam Proton 10.0 beta. So far so good running out of the box without any tweaks! Seems smoother than before as well.

**Comment by OlivierFRDierick** [comment]
Hello,

I no longer have "stack smashing detected" error.
The game loads fine again.

I had a crash the first time, when being presented the new game event(?) with the robot, but after force-closing the game and restarting, I could reach the garage and do a test drive without issue.

Regards.

**Comment by FlareFlo** [comment]
I too can attest that the issue is no longer present, game runs perfectly fine now.
